### PR TITLE
Introduce disk utilization error handler

### PR DESCRIPTION
**Pull request by AleksanderWWW**
## Before submitting checklist

- [x] Did you **update the CHANGELOG**? (not for test updates, internal changes/refactors or CI/CD setup)
- [x] Did you **ask the docs owner** to review all the user-facing changes?

### CODE REVIEW

**Comment by AleksanderWWW** [comment]
@normandy7 Would need you to take a look at naming and error/warning messages :smile: Let me know if you have any questions!

File: src/neptune/exceptions.py [hunk 1]
```diff
@@ -1215,3 +1216,18 @@ def __init__(self, attribute: str, path: str):
...
         super().__init__(message.format(path=path_to_str([attribute, path]), **STYLES))
+
+
+class NeptuneMaxDiskUtilizationExceeded(NeptuneException):
+    def __init__(self, disk_utilization: float, utilization_limit: float):
+        message = """
+    {h1}
+    ----NeptuneMaxDiskUtilizationExceeded----------------------------------------------
+    {end}
+    Current disk utilization ({disk_utilization}) exceeds the limit ({utilization_limit}).
```
Comment (normandy7) [inline src/neptune/exceptions.py:1227]: If these are percentages, maybe it's helpful to show percentage signs?

```suggestion
    Current disk utilization ({disk_utilization}%) exceeds the limit ({utilization_limit}%).
```

File: src/neptune/exceptions.py [hunk 2]
```diff
@@ -1215,3 +1216,18 @@ def __init__(self, attribute: str, path: str):
         {correct}Need help?{end}-> https://docs.neptune.ai/getting_help/
         """
         super().__init__(message.format(path=path_to_str([attribute, path]), **STYLES))
+
+
+class NeptuneMaxDiskUtilizationExceeded(NeptuneException):
+    def __init__(self, disk_utilization: float, utilization_limit: float):
+        message = """
+    {h1}
+    ----NeptuneMaxDiskUtilizationExceeded----------------------------------------------
```
Comment (Raalsky) [inline src/neptune/exceptions.py:1225]: Remove ident. Screenshot:
Screenshot 2023-11-09 at 15 42 14

File: src/neptune/internal/utils/disk_utilization.py [hunk 3]
```diff
@@ -75,36 +78,100 @@ def get_max_disk_utilization_from_env() -> Optional[float]:
...
+        return self.func(*self.args, **self.kwargs)
+
+    def handle_utilization_calculation_error(self) -> None:
+        return self.func(*self.args, **self.kwargs)
+
+    def handle_limit_not_exceeded(self) -> None:
+        return self.func(*self.args, **self.kwargs)
+
+    def handle_limit_exceeded(self, current_utilization: float) -> None:
+        assert isinstance(self.max_disk_utilization, float)  # for mypy
```
Comment (Raalsky) [inline src/neptune/internal/utils/disk_utilization.py:159]: How about `if isinstance(...)`?

**Review by Raalsky — APPROVED** [review]

**Pull request merged**